cpp/include/raft/sparse/matrix/diagonal.cuh

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <raft/core/device_csr_matrix.hpp>
+#include <raft/core/device_mdspan.hpp>
+#include <raft/core/resource/cuda_stream.hpp>
+#include <raft/matrix/init.cuh>
+#include <raft/sparse/matrix/detail/diagonal.cuh>
+#include <raft/util/input_validation.hpp>
+
+namespace raft::sparse::matrix {
+
+/**
+ * @brief Get the diagonal vector from a CSR matrix
+ *
+ * This function extracts the diagonal elements from a CSR matrix and stores them in a vector.
+ * The diagonal elements are the elements where the row index and column index are the same.
+ *
+ *
+ * @tparam T The data type of the matrix elements
+ * @tparam IndexType The data type of the indices
+ *
+ * @param[in]  res             RAFT resources for managing device memory and streams
+ * @param[in]  csr_matrix_view View of the input CSR matrix from which to extract the diagonal
+ * @param[out] diagonal        View of the output vector where diagonal elements will be stored
+ *
+ */
+template <typename T, typename IndexType>
+void get_diagonal_vector_from_csr(
```

Comment:
Please rename to `diagonal`. CSR should be implicit in the argument list (and docs).